### PR TITLE
Support for reading/writing shallow envelope protobuf messages

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: writer/writer.go [hunk 1]
```diff
@@ -157,18 +172,92 @@ func generateWriteValue(data []byte, writeOpts *opts.WriteOptions, md *desc.Mess
...
+	pbBytes, err := proto.Marshal(envelope)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to marshal dynamic protobuf message to bytes")
+	}
+
+	return pbBytes, nil
+}
+
+// convertJSONPBToProtobufShallow converts a JSONPB message to a protobuf message using shallow envelope schemas
+func convertJSONPBToProtobufShallow(data []byte, envelope, payload *dynamic.Message, payloadFieldID int32) ([]byte, error) {
```
Comment (dselans) [inline writer/writer.go:200]: LOL this is rough. Thanks for getting through it @blinktag !

**Review by dselans — APPROVED** [review]
Amazing work @blinktag - looks great! All minor comments.

File: docs/examples.md [hunk 2]
```diff
@@ -496,19 +498,49 @@ slice to the message bus.
...
+the message itself, and another type of message used to decode the protobuf contents of a field inside the envelope.
+
+To read/write shallow envelope messages with plumber, you will need to specify the following additional flags:
+
+1. `--protobuf-envelope-type shallow` - To indicate that the message is a shallow envelope
+2. `--shallow-envelope-field-number` - Protobuf field number of the envelope's field which contains the encoded protobuf data
+3. `--shallow-envelope-message` - The protobuf message name used to encode the data in the field
+
+##### Writing shallow envelope
+```bash
```
[the remark this answers is not in the document]
Comment (blinktag) [reply docs/examples.md:539]: Done!

File: docs/examples.md [hunk 3]
```diff
@@ -496,19 +498,49 @@ slice to the message bus.
...
   --protobuf-root-message pkg.Message \
   --input-file ~/fakes/some-jsonpb-file.json \ 
   --encode-type jsonpb
 ```
 
-##### Using Avro schemas when reading or writing
+#### Shallow envelope protobuf messages
+
+Plumber supports "shallow envelope" protobuf messages consisting of one type of protobuf message used to decode
+the message itself, and another type of message used to decode the protobuf contents of a field inside the envelope.
```
[the remark this answers is not in the document]
Comment (blinktag) [reply docs/examples.md]: Done!

**Pull request merged**